### PR TITLE
Feature/e picture

### PR DESCRIPTION
**Pull request by patric-eberle**
Adds an e-picture element which allows to insert a responsive `` (multiple sizes with same ratio) or `<picture>` (multiple sizes and ratios) element.

Ticket: https://winterhalter-fenner.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=WF&modal=detail&selectedIssue=WF-445

Documentation: http://localhost:6060/styleguide#!/e-picture
Lazyload Demo: http://localhost:8080/styleguide/lazyload

### CODE REVIEW

File: app/components/e-picture.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        200: 'http://via.placeholder.com/200x100',
+        400: 'http://via.placeholder.com/400x200',
+        800: 'http://via.placeholder.com/800x400',
+        1400: 'http://via.placeholder.com/1400x700',
+      },
+      sizes: {
+        1024: 1400,
+        xs: 200,
+        md: 400,
+      },
```
Comment (geroldboehler) [inline app/components/e-picture.md]: Use larger sizes

File: app/components/e-picture.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        800: ['http://via.placeholder.com/800x150/aaaaaa 1x', 'http://via.placeholder.com/1600x300/aaaaaa 2x'],
+        1440: ['http://via.placeholder.com/1440x150/dddddd 1x', 'http://via.placeholder.com/2880x300/dddddd 2x'],
+      },
+    })
+  };
+</script>
+```
+
+#### Img element with srcset and sizes
+```
```
Comment (geroldboehler) [inline app/components/e-picture.md]: Add explanation why image won't change when downsizing the window.

File: app/components/e-picture.vue [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+      /**
+       * A list of source definitions
+       *
+       * `{ <breakpoint (px value|short name)>: [<url> & <density>], ... }`
+       */
+      sources: {
+        type: Object,
+        default: null,
+      },
+      lazyload: { // Note: breaks srcset in IE11, because picturefill will be initialized AFTER the fallback has been loaded
```
Comment (geroldboehler) [inline app/components/e-picture.vue]: Prüfen

File: app/components/e-picture.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        200: 'http://via.placeholder.com/100x50',
+        400: 'http://via.placeholder.com/200x100',
+        800: 'http://via.placeholder.com/300x150',
+        1400: 'http://via.placeholder.com/600x300',
+      },
+      sizes: {
+        1024: 600,
+        xs: 100,
+        md: 300,
+      },
```
Comment (geroldboehler) [inline app/components/e-picture.md]: Use larger sizes

File: app/components/e-picture.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
```
Comment (geroldboehler) [inline app/components/e-picture.md]: Use larger sizes

File: app/components/e-picture.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+#### Preserve ratio inline (tinted background for visualisation)
+```
+<template>
+  <div>
+    <e-picture
+      :sizes="sizes"
+      :srcset="srcset"
+      :fallback="fallback"
+      :ratio="800/800"
```
Comment (geroldboehler) [inline app/components/e-picture.md]: Use bitter ratio so that the blue box is not as big

File: app/components/e-picture.md (hunk 2 above)
Comment (patric-eberle) [reply app/components/e-picture.md]: Done

**Comment by patric-eberle** [comment]
@geroldboehler Updated the demo and breakpoints. Please check again.

File: app/components/e-picture.vue [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+              throw new Error('Invalid breakpoint value for e-picture source.');
+            }
+
+            return {
+              sort: parseInt(breakpointWidth, 10),
+              media: `(min-width: ${breakpointWidth}px)`,
+              srcset: this.sources[breakpoint].join(',')
+            };
+          })
+          .sort((a, b) => (a.sort < b.sort ? 1 : -1));
```
Comment (geroldboehler) [inline app/components/e-picture.vue]: Deactivate linter error message

Comment (patric-eberle) [reply app/components/e-picture.vue]: Done.

**Pull request merged**